src/software/ai/hl/stp/tactic/pass_defender/pass_defender_fsm.h

```diff
@@ -82,17 +117,27 @@ struct PassDefenderFSM
         DEFINE_SML_ACTION(blockPass)
         DEFINE_SML_ACTION(interceptBall)
 
+        DEFINE_SML_STATE(DribbleFSM)
+        DEFINE_SML_GUARD(ballNearbyWithoutThreat)
+        DEFINE_SML_SUB_FSM_UPDATE_ACTION(prepareGetPossession, DribbleFSM)
+
         return make_transition_table(
             // src_state + event [guard] / action = dest_state
             *BlockPassState_S + Update_E[passStarted_G] / interceptBall_A =
                 InterceptBallState_S,
             BlockPassState_S + Update_E / blockPass_A,
             InterceptBallState_S + Update_E[ballDeflected_G] / blockPass_A =
                 BlockPassState_S,
+            InterceptBallState_S + Update_E[ballNearbyWithoutThreat_G] /
```

Comment:
should we also have a transition to `DribbleFSM` from `BlockPassState`

Reply:
I guess we are trying to gain possessions of balls that we intercept, so this is out-of-scope for this PR

Reply:
Oh yes, I originally implemented this feature, but found that it wrecked havoc on the current FSM tests. I think we can fix this issue in another future PR targeted at making the Defenders more aggressive.